server/src/main/java/com/paypal/selion/proxy/SeLionSauceProxy.java

```diff
@@ -23,71 +23,91 @@
 import com.paypal.selion.utils.SauceLabsRestApi;
 
 import org.apache.commons.lang.StringUtils;
 import org.openqa.grid.common.RegistrationRequest;
 import org.openqa.grid.common.exception.GridConfigurationException;
 import org.openqa.grid.common.exception.GridException;
-import org.openqa.grid.internal.BaseRemoteProxy;
 import org.openqa.grid.internal.Registry;
 import org.openqa.grid.internal.TestSession;
+import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
 
 import com.paypal.selion.logging.SeLionGridLogger;
 
 /**
- * This customized version of {@link BaseRemoteProxy} basically helps redirect all traffic to the SauceLabs cloud.
+ * This customized version of {@link DefaultRemoteProxy} basically helps redirect all traffic to the SauceLabs cloud.
  */
-public class SeLionSauceProxy extends BaseRemoteProxy {
+public class SeLionSauceProxy extends DefaultRemoteProxy {
 
     private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(SeLionSauceProxy.class);
+
+    private static final String DEFAULT_TO_USERNAME_TOKEN = "$username";
+    private static final String PARENT_TUNNEL_CAPABILITY = "parent-tunnel";
+    private static final String USERNAME_CAPABILITY = "username";
+
     private final SauceLabsRestApi sauceApi;
+    private final String defaultParentTunnel;
     private final Lock accessLock = new ReentrantLock();
 
     public SeLionSauceProxy(RegistrationRequest request, Registry registry) {
         super(request, registry);
 
-        if (! SauceServlet.PROXY_ID.equals(remoteHost.toExternalForm())) {
+        if (!SauceServlet.PROXY_ID.equals(remoteHost.toExternalForm())) {
             throw new GridException(SeLionSauceProxy.class.getSimpleName() + " can not be used by an external process");
         }
 
         try {
+            defaultParentTunnel = SauceConfigReader.getInstance().getDefaultParentTunnel();
             sauceApi = new SauceLabsRestApi();
-        } catch (GridConfigurationException e ) {
+        } catch (GridConfigurationException e) {
             throw new GridException("Failed to initialize proxy: ", e);
         }
     }
 
     @Override
-    public TestSession getNewSession(Map<String, Object> requestedCapability) {
+    public TestSession getNewSession(Map<String, Object> requestedCapabilities) {
+        LOGGER.entering();
         TestSession session = null;
         try {
             accessLock.lock();
 
             try {
+                // TODO address this superficial wait for sauce labs. Can we do better?
                 Thread.sleep(3000);
             } catch (InterruptedException e) {
                 LOGGER.log(Level.SEVERE, e.getMessage(), e);
             }
+
+            // setup the parent-tunnel, if required
+            processParentTunnelCapability(requestedCapabilities);
+
             if (sauceApi.getNumberOfTCRunning() <= getMaxNumberOfConcurrentTestSessions()) {
-                String username = (String) requestedCapability.get("sauceUserName");
```

Comment:
Nice call on removing this unnecessary conversion and make the capabilties key match what sauce is expecting.